src/wallet/rpcwallet.cpp

```diff
@@ -2577,23 +2577,50 @@ UniValue z_listunspent(const UniValue& params, bool fHelp)
         obj.pushKV("confirmations", entry.confirmations);
         bool hasSaplingSpendingKey = pwalletMain->HaveSaplingSpendingKeyForAddress(entry.address);
         obj.pushKV("spendable", hasSaplingSpendingKey);
-        obj.pushKV("address", keyIO.EncodePaymentAddress([&]() {
-            auto ua = pwalletMain->FindUnifiedAddressByReceiver(entry.address);
-            if (ua.has_value()) {
-                return libzcash::PaymentAddress{ua.value()};
-            } else {
-                return libzcash::PaymentAddress{entry.address};
-            }
-        }()));
+        auto addr = pwalletMain->GetPaymentAddressForRecipient(entry.op.hash, entry.address);
+        if (addr.second != RecipientType::WalletInternalAddress) {
+            obj.pushKV("address", keyIO.EncodePaymentAddress(addr.first));
+        }
         obj.pushKV("amount", ValueFromAmount(CAmount(entry.note.value()))); // note.value() is equivalent to plaintext.value()
         obj.pushKV("memo", HexStr(entry.memo));
         if (hasSaplingSpendingKey) {
-            obj.pushKV("change", pwalletMain->IsNoteSaplingChange(saplingNullifiers, entry.address, entry.op));
+            obj.pushKV(
+                    "change",
+                    pwalletMain->IsInternalRecipient(entry.address) ||
+                    pwalletMain->IsNoteSaplingChange(saplingNullifiers, entry.address, entry.op));
         }
         results.push_back(obj);
     }
 
-    // TODO ORCHARD #5683
+    for (auto & entry : orchardEntries) {
+        UniValue obj(UniValue::VOBJ);
+        obj.pushKV("txid", entry.GetOutPoint().hash.ToString());
+        obj.pushKV("pool", ADDR_TYPE_ORCHARD);
+        obj.pushKV("type", ADDR_TYPE_ORCHARD); //deprecated
```

Comment:
```suggestion
```
This is not needed for compatibility since it was never present before, unlike in the Sapling case.